File: 0021-merge-two-sorted-lists-合并两个有序链表/0021.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        bool isBreak = false;
        ListNode* result = new ListNode(0);
        ListNode* temp = result;
        while(l1||l2){
            if(l1 && l2){
                if(l1->val<=l2->val){
                    temp->next = new ListNode(l1->val);
                    temp=temp->next;
                    l1=l1->next;
                }else{
                    temp->next = new ListNode(l2->val);
                    temp=temp->next;
                    l2=l2->next;
                }
            }else{
                if(l1){
                    temp->next = new ListNode(l1->val);
                    temp=temp->next;
                    l1=l1->next;
                }
                if(l2){
                    temp->next = new ListNode(l2->val);
                    temp=temp->next;
                    l2=l2->next;
                }
            }
        }return result->next;
    }
};

File: 0021-merge-two-sorted-lists-合并两个有序链表/0021.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        rtn = ListNode(0)
        l = rtn
        p1 = l1
        p2 = l2
        while p1 and p2:
            if p1.val < p2.val:
                l.next = p1
                p1 = p1.next
            else:
                l.next = p2
                p2 = p2.next
            l = l.next
        
        while p1:
            l.next = p1
            p1 = p1.next
            l = l.next

        while p2:
            l.next = p2
            p2 = p2.next
            l = l.next

        return rtn.next